src/solace_ai_connector/components/general/langchain/langchain_embeddings.py

```diff
@@ -66,13 +66,28 @@ def __init__(self, **kwargs):
         super().__init__(info, **kwargs)
 
     def invoke(self, message, data):
-        text = data["text"]
+        items = data["items"]
         embedding_type = data.get("type", "document")
 
-        embeddings = None
+        items = [items] if type(items) != list else items
+
         if embedding_type == "document":
-            embeddings = self.component.embed_documents([text])
+            return self.embed_documents(items)
         elif embedding_type == "query":
-            embeddings = [self.component.embed_query(text)]
+            return self.embed_queries(items)
+        elif embedding_type == "image":
+            return self.embed_images(items)
+
+    def embed_documents(self, documents):
+        embeddings = self.component.embed_documents(documents)
+        return {"embeddings": embeddings}
+    
+    def embed_queries(self, queries):
+        embeddings = []
+        for query in queries:
+            embeddings.append(self.component.embed_query(query))
+        return {"embeddings": embeddings}
```

Comment:
LangChain doesn't expose the function to do batch in single calls, so had to loop here